controllers/machine_controller_test.go

```diff
@@ -455,6 +456,162 @@ func TestReconcileRequest(t *testing.T) {
 	}
 }
 
+func TestMachineReadyConditionSummary(t *testing.T) {
```

Comment:
From my understanding this test it is verifying two things at the same time:
1. The controller applies fallback conditions if the ready conditions are missing from infra or bootstrap objects
2. The controller computes condition summary respecting priority.

In order to improve readability/maintainability, what about having two separate tests, one for each of the above points?
Also, in the second test, should we set the ready condition, so it is more simple to identify the expected result when computing the summary (without being dragged into the fallback conditions)?

Reply:
Updated the tests to assert the conditions for the different scenarios